src/ScheduleFunctions.cpp

```diff
@@ -534,6 +540,26 @@ Stmt build_produce(Function f) {
             }
         }
 
+        Stmt annotate;
+        if (target.has_feature(Target::MSAN)) {
+            // Mark the buffers as initialized before calling out.
```

Comment:
If you agree with the comments above, then the body of this if could just become

    annotate = Block::make(annotate_stmts);

(where annotate_stmts is the vector containing the annotate calls)